ignition/service/infrastructure.py

```diff
@@ -267,11 +292,28 @@ def __init__(self, **kwargs):
             if 'inf_monitor_service' not in kwargs:
                 raise ValueError('inf_monitor_service argument not provided (required when async_messaging_enabled is True)')
             self.inf_monitor_service = kwargs.get('inf_monitor_service')
+        self.async_requests_enabled = infrastructure_config.request_queue.enabled
+        if self.async_requests_enabled:
+            if 'request_queue' not in kwargs:
+                raise ValueError('request_queue argument not provided (required when async_requests_enabled is True)')
+            self.request_queue = kwargs.get('request_queue')
 
     def create_infrastructure(self, template, template_type, system_properties, properties, deployment_location):
-        create_response = self.driver.create_infrastructure(template, template_type, system_properties, properties, deployment_location)
-        if self.async_enabled is True:
-            self.__async_infrastructure_task_completion(create_response.infrastructure_id, create_response.request_id, deployment_location)
+        if self.async_requests_enabled:
+            self.request_queue.queue_infrastructure_request({
+                # 'infrastructure_id': create_response.infrastructure_id,
+                'request_id': str(uuid.uuid4()),
+                'template': template,
+                'template_type': template_type,
+                'properties': properties,
+                'system_properties': system_properties,
+                'deployment_location': deployment_location
+            })
+        else:
+            create_response = self.driver.create_infrastructure(template, template_type, PropValueMap(system_properties), PropValueMap(properties), deployment_location)
+            if self.async_enabled is True:
+                self.__async_infrastructure_task_completion(create_response.infrastructure_id, create_response.request_id, deployment_location)
+
         return create_response
```

Comment:
if self.async_requests_enabled==True won't the create_response be None?

Reply:
Yep, I'll fix this up. Good spot!